modules/core/rapid_context.js

```diff
@@ -1,34 +1,73 @@
-import { geoExtent } from '../geo';
+import { geoExtent, geoExtentFromBounds } from '../geo';
 
 import toGeoJSON from '@mapbox/togeojson';
-
+import { dispatch as d3_dispatch } from 'd3-dispatch'; 
+import { utilRebind } from '../util'
 
 export function coreRapidContext(context) {
     var rapidContext = {};
     rapidContext.version = '1.0.1';
+    var _isRect = undefined; 
+    var dispatch = d3_dispatch('task_extent_set');
+
+    function distinct (value, index, self) {
+        return self.indexOf(value) === index; 
+    }
 
     var taskExtent;
     rapidContext.setTaskExtentByGpxData = function(gpxData) {
         var dom = (new DOMParser()).parseFromString(gpxData, 'text/xml');
         var gj = toGeoJSON.gpx(dom);
+        var uniqueLats = []; 
+        var uniqueLngs = []; 
         if (gj.type === 'FeatureCollection') {
             var minlat, minlon, maxlat, maxlon;
+            //Calculate task extent. 
             gj.features.forEach(function(f) {
                 if (f.geometry.type === 'Point') {
                     var lon = f.geometry.coordinates[0];
                     var lat = f.geometry.coordinates[1];
                     if (minlat === undefined || lat < minlat) minlat = lat;
                     if (minlon === undefined || lon < minlon) minlon = lon;
                     if (maxlat === undefined || lat > maxlat) maxlat = lat;
-                    if (maxlon === undefined || lon > maxlon) maxlon = lon;
+                    if (maxlon === undefined || lon > maxlon) maxlon = lon;                    
+                }
+    
+                if (f.geometry.type === 'LineString') {
```

Comment:
Perhaps I should leave a comment here explaining that the 'LineString' portion of the task geoJson explicitly defines just the task bbox. 

Actually, @gaoxm @zlavergne  is that a valid assumption to make? Are there other formats to consider for task bboxes/gpx urls?

Reply:
not sure what could possibly be returned by TM - I think we should also allow 'Polygon'. @zlavergne could you help check what could be returned by TM?

Reply:
Okay, I did some further digging and it does look like I need to keep the 'LineString' constraint here. Here's why- the .gpx data fed to us by TM also contains points that show the task id/ state. These points do not necessarily share the same lats/lngs as the four corners of the task box- sometimes the points will be along the interior of a line segment.